substrate-query-framework/index-builder/src/indexer/IndexerStatusService.ts

```diff
@@ -36,63 +31,98 @@ export class IndexerStatusService {
     .catch((e) => { throw new Error(e) });
     
     this.redisSub.on('message', (channel, message) => {
-      debug(`Got message: ${message as string}`);
-      if (channel === BLOCK_COMPLETE_CHANNEL) {
-        this.onCompleteBlock((JSON.parse(message) as BlockPayload).height)
-          .catch((e) => { throw new Error(`Error connecting to Redis: ${logError(e)}`) });
-      }
+      this.onNewMessage(channel, message)
+          .catch((e) => { throw new Error(`Error connecting to Redis: ${logError(e)}`) })
     })
   }  
   
+  async onNewMessage(channel: string, message: string): Promise<void> {
+    if (channel === BLOCK_COMPLETE_CHANNEL) {
+      const payload = JSON.parse(message) as BlockPayload;
```

Comment:
Is there a chance to fail?

Reply:
So far there's a single publisher to this channel, so it's fairly safe. If it fails, an error will be thrown and logged, which would be easy to spot.